src/main/java/com/google/sps/servlets/ReviewsServlet.java

```diff
@@ -53,17 +54,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) {
 
       // Create review entity and get its key.
       Entity reviewEntity = new Entity("Review");
-      reviewEntity.setProperty("userID", userID);
-      reviewEntity.setProperty("comment", comment);
-      reviewEntity.setProperty("rating", rating);
-      reviewEntity.setProperty("dateTime", dateTime);
+      reviewEntity.setProperty(REVIEW_USERID, userID);
+      reviewEntity.setProperty(REVIEW_COMMENT, comment);
+      reviewEntity.setProperty(REVIEW_RATING, rating);
+      reviewEntity.setProperty(REVIEW_DATETIME, dateTime);
       datastore.put(reviewEntity);
       Key reviewKey = reviewEntity.getKey();
 
       // Append key to its Business entity.
       reviewsKeyList.add(reviewKey);
       businessEntity.setProperty(BUSINESS_REVIEWS, gson.toJson(reviewsKeyList));
 
+      // Recalculate and update aggregated rating.
+      float aggregatedRating = recalculateRating(businessEntity, reviewsKeyList.size(), rating);
+      businessEntity.setProperty(BUSINESS_RATING, aggregatedRating);
+
       // Redirect back to the product page.
       response.sendRedirect("/product.html?businessID="+businessKey);
```

Comment:
Consider using template literals or String formatting. 
`String.format("/product.html?businessID=%s", businessKey);`

Reply:
also, businessKey is a Key type, but this should be parsed to a Long. `businessKey.getId()`

Reply:
i don't seee businessKey initialised anywhere in this file?

Reply:
line 49, type Key!

Reply:
@jirehcwe Thanks for catching that. I edited it to use the original parameter parsed as a Long for the redirect (it's still stored as Key in reviewsKeyList).
@dalisc It was assigned in line 50 as a Key type!